docker/nobinary.txt

```diff
@@ -1 +1,5 @@
-# Empty for now
+# libraries to compile with local gdal
+--no-binary rasterio
+--no-binary fiona
+--no-binary shapely
+--no-binary cf-units
```

Comment:
`pyproj`?

Reply:
Pyproj seems to be installing fine from pip without `--no-binary`.

The problem I'm currently stuck on is that rasterio can't read netcdf files:

```
>>> import rasterio
>>> rasterio.drivers.driver_from_extension("multi_doc.nc")
Traceback (most recent call last):
  File "/env/lib/python3.12/site-packages/rasterio/drivers.py", line 71, in driver_from_extension
    return driver_extensions[os.path.splitext(path)[-1].lstrip(".").lower()]
           ~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
KeyError: 'nc'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/env/lib/python3.12/site-packages/rasterio/drivers.py", line 73, in driver_from_extension
    raise ValueError("Unable to detect driver. Please specify driver.")
ValueError: Unable to detect driver. Please specify driver.
>>>
```

The docker image has `libnetcdf-dev` installed with apt and `netcdf==1.6.2` installed with pip. 

I don't suppose you've got any suggestions for what might be missing for that?

Reply:
my suggestion is don't bother with non-conda installs, but this looks like your gdal is missing netcdf support OR your build stage has more libs than you use stage.

conda envs are somewhat larger than pip, but at least you don't have that nightmare to deal with. Alternatively go all out and build EVERYTHING from source and stop mixing pre-compiled things from pip with locally compiled code.

Reply:
Now I've got it working we'll probably stick with this for the short term.  In the longer term, @omad has ideas for a long overdue complete restructure the way tests are run in github (including dropping docker all together).  An Issue will be created soon.